File: php/src/MDS.php
<?php

function lox617_gf_mul(
    int $a,
    int $b
): int {

    $p = 0;

    for($i = 0; $i < 8; $i++) {

        if($b & 1) {
            $p ^= $a;
        }

        $hi =
            $a & 0x80;

        $a =
            ($a << 1) & 0xFF;

        if($hi) {
            $a ^= 0x1B;
        }

        $b >>= 1;
    }

    return $p;
}

function lox617_mds_apply(
    array &$block
): void {

    $mds = [
        0x17,0x31,0x59,0x83,
        0xA7,0xC1,0xD3,0xEF
    ];

    $temp =
        array_fill(
            0,
            8,
            0
        );

    for($i = 0; $i < 8; $i++) {

        for($j = 0; $j < 8; $j++) {

            $temp[$i] ^=
                lox617_gf_mul(
                    $block[$j],
                    $mds[
                        ($i + $j) % 8
                    ]
                );
        }
    }

    for($i = 0; $i < 8; $i++) {

        $block[$i] =
            $temp[$i] & 0xFF;
    }
}

File: rust/src/mds.rs
pub fn lox_gf256_mul(
    mut a: u8,
    mut b: u8
) -> u8 {

    let mut p = 0u8;

    for _ in 0..8 {

        if (b & 1) != 0 {
            p ^= a;
        }

        let hi =
            a & 0x80;

        a <<= 1;

        if hi != 0 {
            a ^= 0x1B;
        }

        b >>= 1;
    }

    p
}

pub fn lox617_mds_apply(
    block: &mut [u8; 8]
) {

    let mds: [u8; 8] = [
        0x17, 0x31,
        0x59, 0x83,
        0xA7, 0xC1,
        0xD3, 0xEF
    ];

    let mut temp =
        [0u8; 8];

    for i in 0..8 {

        for j in 0..8 {

            temp[i] ^=
                lox_gf256_mul(
                    block[j],
                    mds[
                        (i + j) % 8
                    ]
                );
        }
    }

    *block = temp;
}